ClosedXML/Excel/CalcEngine/Expression.cs

```diff
@@ -10,7 +10,18 @@ namespace ClosedXML.Excel.CalcEngine
 {
     internal abstract class ExpressionBase
     {
+        private CalculationContext _context;
+        protected ExpressionBase(in CalculationContext ctx)
+        {
+            _context = ctx;
+        }
+
         public abstract string LastParseItem { get; }
+
+        public ref CalculationContext CurrentContext
```

Comment:
I would like to move toward stateless parser and calculation engine, though I am not sure if it is acheivable. 
This would, theoretically, allow doing calculations in parallel eventually.